Mathlib/RingTheory/Adjoin/Basic.lean

```diff
@@ -480,4 +480,55 @@ def Subring.closureEquivAdjoinInt {R : Type*} [Ring R] (s : Set R) :
     Subring.closure s ≃ₐ[ℤ] Algebra.adjoin ℤ s :=
   Subalgebra.equivOfEq (subalgebraOfSubring <| Subring.closure s) _ (adjoin_int s).symm
 
-  end NatInt
+end NatInt
+
+/-- If `K / E / F` is a ring extension tower, `L` is a subalgebra of `K / F` which is generated by
+`S` as an `F`-module, then `E[L]` is generated by `S` as an `E`-module. -/
+theorem Subalgebra.adjoin_eq_span_of_eq_span {F : Type*} (E : Type*) {K : Type*}
+    [CommSemiring F] [CommSemiring E] [CommSemiring K]
+    [Algebra F E] [Algebra E K] [Algebra F K] [IsScalarTower F E K]
+    (L : Subalgebra F K) {S : Set K}
+    (h : Subalgebra.toSubmodule L = Submodule.span F S) :
+    Subalgebra.toSubmodule (Algebra.adjoin E (L : Set K)) = Submodule.span E S := by
+  have h0 := h.symm ▸ Submodule.span_le_restrictScalars F E S
+  change (L : Set K) ⊆ Submodule.span E S at h0
```

Comment:
```suggestion
  have h0 : (L : Set K) ⊆ Submodule.span E S := h.symm ▸ Submodule.span_le_restrictScalars F E S
```
(untested)

Reply:
No, that doesn't work. I don't know why, maybe `▸` doesn't like this syntax.